token/program/src/processor.rs

```diff
@@ -207,6 +203,12 @@ impl Processor {
             )?,
         };
 
+        // This check MUST occur just before the amounts are manipulated
```

Comment:
Is this even needed?   It looks like the code that follows will handle the self-transfers, so having a single code path for simplicity seems worth the slight overhead this would cause for a self-transfer

Reply:
Yeah, see explanation in the description